h2o-core/src/main/java/hex/ModelMetricsBinomial.java

```
@@ -167,37 +169,62 @@ static public ModelMetricsBinomial make(Vec targetClassProbs, Vec actualLabels,
+                  : frameWithWeights.vec(m._parms._response_column);
+          if (resp != null) {
+            Vec weight = m==null?null : frameWithWeights.vec(m._parms._weights_column);
+            gl = calculateGainsLift(m, preds, resp, weight);
+          }
+        }
+      }
+      return makeModelMetrics(m, f, gl);
+    }
+
+    private ModelMetrics makeModelMetrics(Model m, Frame f, GainsLift gl) {
       double mse = Double.NaN;
       double logloss = Double.NaN;
       double sigma = Double.NaN;
       AUC2 auc = null;
-      GainsLift gl = null;
       if (_wcount > 0) {
         sigma = weightedSigma();
         mse = _sumsqe / _wcount;
         logloss = _logloss / _wcount;
         auc = new AUC2(_auc);
-        gl = null;
-        if (preds!=null) {
-          Vec resp = m==null && f.vec(f.numCols()-1).isCategorical() ? f.vec(f.numCols()-1) //work-around for the case where we don't have a model, assume that the last column is the actual response
-                  : f.vec(m._parms._response_column);
-          Vec weight = m==null?null : frameWithWeights.vec(m._parms._weights_column);
-          if (resp != null) {
-            try {
-              gl = new GainsLift(preds.lastVec(), resp, weight);
-              gl.exec(m != null ? m._output._job : null);
-            } catch(Throwable t) {}
-          }
-        }
       }
       ModelMetricsBinomial mm = new ModelMetricsBinomial(m, f, _count, mse, _domain, sigma, auc,  logloss, gl);
       if (m!=null) m.addModelMetrics(mm);
       return mm;
     }
+
+    private GainsLift calculateGainsLift(Model m, Frame preds, Vec resp, Vec weights) {
+      GainsLift gl = null;
+      try {
+        gl = new GainsLift(preds.lastVec(), resp, weights);
+        gl.exec(m != null ? m._output._job : null);
+      } catch(Throwable t) { // TODO: Why do we need to catch Throwable here?
```

I would change it to Exception, at least we will not hide any Errors